### PR TITLE
PWN-8470 - fixed user tokens updating

### PR DESCRIPTION
**Pull request by eduardmaximovich**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-8470

## Description of Work

- Fixed data race while updating balances from multiple sources at once

### CODE REVIEW

**Review by kabramovp2p — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/home/repository/HomeLocalRepository.kt [hunk 1]
```diff
@@ -9,6 +9,7 @@ interface HomeLocalRepository {
     suspend fun removeIfExists(publicKey: String, symbol: String)
     fun getTokensFlow(): Flow<List<Token.Active>>
     suspend fun getUserTokens(): List<Token.Active>
+    suspend fun findUserTokenByMint(mintAddress: String): Token.Active?
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/repository/HomeLocalRepository.kt:12]: use Base58String

Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/home/repository/HomeLocalRepository.kt:12]: and could you put this method to UserTokensRepository? 
HomeLocal name doesn't tell us anything about user tokens

File: app/src/main/java/org/p2p/wallet/home/repository/UserTokensLocalRepository.kt [hunk 2]
```diff
@@ -49,28 +49,13 @@ class UserTokensLocalRepository(
         tokenMint: Base58String,
         accountPublicKey: Base58String
     ) {
-        // we modify this list, it's more productive then do .map over N elements
-        val cachedTokens = homeLocalRepository.getUserTokens().toMutableList()
+        val token = homeLocalRepository.findUserTokenByMint(tokenMint.base58Value)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/repository/UserTokensLocalRepository.kt:52]: nice!!

Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/home/repository/UserTokensLocalRepository.kt:52]: tokenToUpdate maybe?

File: app/src/main/java/org/p2p/wallet/home/ui/main/HomePresenter.kt [hunk 3]
```diff
@@ -211,9 +191,18 @@ class HomePresenter(
...
+        homeStateSubscribed = true
+
         tokensPolling.shareTokenPollFlowIn(this)
             .filterNotNull()
+            .onCompletion { homeStateSubscribed = false }
             .collect { homeState ->
+                run {
+                    val solTokensLog = homeState.solTokens
+                        .joinToString { "${it.tokenSymbol}(${it.total.formatToken()}; ${it.totalInUsd?.formatFiat()})" }
+                    Timber.i("Home state solTokens: $solTokensLog")
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/ui/main/HomePresenter.kt:204]: Timber.d i think, no need to put this info to Sentry

**Review by gslevinkov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/home/repository/HomeLocalRepository.kt (hunk 1 above)
Comment (eduardmaximovich) [reply app/src/main/java/org/p2p/wallet/home/repository/HomeLocalRepository.kt:12]: @gslevinkov I haven't understood your point. HomeLocalRepository already has `getUserTokens` and `getUserBalance` methods. I could rename these methods to `getTokens` and `getBalance` respectively, but would that make sense?

**Pull request merged**